connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -412,14 +412,13 @@ private void rewind() {
         if (offsets.isEmpty()) {
             return;
         }
-        for (TopicPartition tp: offsets.keySet()) {
-            Long offset = offsets.get(tp);
-            if (offset != null) {
-                log.trace("Rewind {} to offset {}.", tp, offset);
-                consumer.seek(tp, offset);
-                lastCommittedOffsets.put(tp, new OffsetAndMetadata(offset));
-                currentOffsets.put(tp, new OffsetAndMetadata(offset));
-            }
+        for (Map.Entry<TopicPartition, Long> entry: offsets.entrySet()) {
+            TopicPartition tp = entry.getKey();
+            Long offset = entry.getValue();
+            log.trace("Rewind {} to offset {}.", tp, offset);
```

Comment:
Actually I agree. We shouldn't remove a check for null values, even if such values shouldn't be stored in the map as type `Long` objects. I was wondering myself and I was sure we would discuss this. I'll reinstate the `if` statement and I'll just add a log message in case we get `null`.